test/rcases.lean

```diff
@@ -227,3 +227,127 @@ example {α n} (h : foo α n) : true := by test_rcases_hint "_ | ⟨n, h_ᾰ⟩"
 
 example {α} (V : set α) (h : ∃ p, p ∈ (V.foo V) ∩ (V.foo V)) :=
 by test_rcases_hint "⟨⟨h_w_fst, h_w_snd⟩, ⟨⟩⟩" 0
+
+section obtain_bug
+
+-- BUG in `obtain`: you can't name hypotheses if you're not destructing/casing at all and
+-- the term is constructed by a tactic block and not a `:=`:
+example : true :=
```

Comment:
This bug took me by surprise. Recorded on #15741 and hopefully fixable soon; I'll take a look at `obtain` in a bit and see if I can figure it out, but currently busy.

Reply:
I think it's to do with line 304-306:

https://github.com/leanprover-community/mathlib/blob/0b377e36ccf5941b57d0885b9173719b97966b56/src/tactic/rcases.lean#L304-L306

This is not true in the trivial case, which is obviously uninteresting from the context of `rcases` and so irrelevant in most cases but will be relevant in the case of `rsufficesI`.